### ISSUE TITLE
add Pyomo 4.0 support

### ISSUE THREAD

**Issue by ojdo** [tum-ens/urbs#13]
Currently, urbs is still stuck with Pyomo support up to version 3.x. As not only the package name was changed from `coopr.pyomo` to `pyomo`, but also some internals for model objects, some more work has to be done on the following interfacing functions:

  * `list_entities`
  * `get_entity`
  * `_get_onset_names`

The remaining code is *probably* unaffected.

**Comment by jinhw1989** [comment]
I would suggest try except import block like:
```python
try:
    from pyomo.core import *
    from pyomo import *
    from pyomo.opt import *
    from pyomo.core.base import *
    from pyomo.environ import *
except:
    # from __future__ import division
    from coopr.pyomo import *
    from coopr.opt import SolverStatus, TerminationCondition, SolutionStatus
    from coopr.opt.base import SolverFactory
    import coopr.environ
``` 
While this will cause efforts to change `pyomo.Set()` into `Set()`.

BTW, this is an interesting project.

**Comment by ojdo** [comment]
@jinhw1989 Thanks for the suggestion! I would rather try to avoid the `*` imports in the try block. I think also that the except block could be avoided by moving to pyomo 4.0 in one swift move without looking back to Coopr. There's always the possiblitiy for "productive" (?) systems to not move along with the new commits/releases.

**Comment by ojdo** [comment]
Closed by commit f79ef4ae90afcf2c649167975c55b07f2831e1fb. It is shorter than the proposed changes by @jinhw1989 and makes the code Python 3 compatible (by removing a call to `iteritems()` and another one to `values()`).

Coopr (aka. Pyomo 3.x) raises a FutureWarning from now on. With with the next/a future release, its support will be dropped.
